apps/examples/src/examples/sync-private-content/SyncPrivateContentExample.tsx

```diff
@@ -0,0 +1,146 @@
+import { useSyncDemo } from '@tldraw/sync'
+import React from 'react'
+import {
+	Atom,
+	TLComponents,
+	Tldraw,
+	react,
+	useAtom,
+	useEditor,
+	useIsToolSelected,
+	useTools,
+	useValue,
+} from 'tldraw'
+import 'tldraw/tldraw.css'
+import { VisibilityOff, VisibilityOn } from '../../icons/icons'
+import { Toggle } from './Toggle'
+import './style.css'
+
+// There's a guide at the bottom of this file!
+
+// [1]
+const PrivateModeContext = React.createContext<null | Atom<boolean>>(null)
+
+const components: TLComponents = {
+	// [2]
+	InFrontOfTheCanvas: () => {
+		const editor = useEditor()
+		const isInSelectTool = useIsToolSelected(useTools().select)
+		const userId = useValue('userId', () => editor.user.getId(), [])
+		const myPrivateSelectedShapes = useValue(
+			'private shapes',
+			() =>
+				editor
+					.getSelectedShapes()
+					.filter((shape) => !!shape.meta.private && shape.meta.ownerId === userId),
+			[editor, userId]
+		)
+
+		// [3]
+		const isPrivateDrawingMode$ = React.useContext(PrivateModeContext)!
+		const isPrivateDrawingMode = useValue(isPrivateDrawingMode$)
```

Comment:
realise the meta is named just `private` which is great, but I'd be tempted to rename this to something more generic, like `PrivateEditingMode` or simple `isPrivateMode` as users aren't always drawing when wanting to work in private mode (can see examples of users editing docs privately before sharing with this sort of functionality at lex).